r10k.gemspec

```diff
@@ -23,7 +23,8 @@ Gem::Specification.new do |s|
   s.license  = 'Apache-2.0'
 
   s.add_dependency 'colored2',   '3.1.2'
-  s.add_dependency 'cri', '2.15.10'
+  s.add_dependency 'cri', '2.15.10' if RUBY_VERSION < '2.5.0'
+  s.add_dependency 'cri' if RUBY_VERSION >= '2.5.0'
```

Comment:
I think we want to at least pin the X version here: `~> 2.15`. I know `cri` has always respected semver, but it still seems good to play it a little safe (and match what we're doing for other gems).

Reply:
updated it

Reply:
I'm wondering if this is a valid thing to do.  ie does this just set the dependency based on what version of ruby is used when `gem build` is run?

Reply:
I want to add that I thought that's safe to use because I saw it earlier on https://github.com/puppetlabs/forge-ruby/blob/main/puppet_forge.gemspec#L37-L38

Reply:
Suggest reverting this PR for now and opening a new discussion if you want to drop ruby 2.4 support.

Reply:
Oh man, this is my bad too, I tested it just with a bundle install and saw that work. Didn't realize the implications for building the gem. Will revert.